File: TelepresenceBot/server-unit/core/clientType.js
module.exports = ClientType = {
    BOT: 'bot',
    HUMAN: 'human',
    TEST: 'test',
    from: function (rawClientType) {
        for (let key in this) {
            if (ClientType[key] == rawClientType) {
                return ClientType[key];
            }
        }
        return undefined;
    }
}

File: TelepresenceBot/mobile/src/main/java/com/novoda/tpbot/support/ClientType.java
package com.novoda.tpbot.support;

public enum ClientType {

    BOT("clientType=bot"),
    HUMAN("clientType=human");

    private final String rawClientType;

    ClientType(String rawClientType) {
        this.rawClientType = rawClientType;
    }

    public String rawClientType() {
        return rawClientType;
    }

}
